p2p/discover/table_test.go

```diff
@@ -396,6 +399,38 @@ func TestTable_revalidateSyncRecord(t *testing.T) {
 	}
 }
 
+// This test checks that ENR filtering is working properly
+func TestTable_filterNode(t *testing.T) {
+	// Create ENR filter
+	type eth struct {
+		ForkID forkid.ID
+		Tail   []rlp.RawValue `rlp:"tail"`
+	}
+
+	enrFilter, _ := ParseEthFilter("bsc")
+
+	// Check test ENR record
+	var r1 enr.Record
+	r1.Set(enr.WithEntry("foo", "bar"))
+	if enrFilter(&r1) {
+		t.Fatalf("filterNode doesn't work correctly for entry")
+	}
+
```

Comment:
Maybe we want to increase the testing logging verbosity. I would add an `t.Logf()` after each successful test step. This helps when reading a failed test.